andreeapetcu/assignment3/TestWordSearch.java

```diff
@@ -0,0 +1,407 @@
+import java.util.*;
+
+public class TestWordSearch {
+
+  private WordSearch wordSearch;
+  private char[][] validGrid;
+  private TrieNode[][] validTrieGrid;
+  private int numberRowsGrid;
+  private int numberColumnsGrid;
+  private Dictionary validDict;
+  private List<String> validListOfWords;
+  private char[][] nullGrid;
+  private Dictionary nullDict;
+  private Dictionary emptyDict;
+  private List<String> emptyListOfDictionaryWords;
+  private boolean[][] validVisited;
+  private boolean[][] nullVisited;
+  private ArrayList<String> validWordsAccumulator;
+  private ArrayList<String> nullWordsAccumulator;
+  private TrieNode validTrieNode;
+  private TrieNode nullTrieNode;
+  private TrieNode[][] nullTrieGrid;
+
+
+  /**
+   * Checks if a list contains a given trie node.
+   *
+   * @param list the list to be searched
+   * @param node the node of interest
+   * @return true if the node is in the list, false otherwise
+   */
+  private boolean containsNode (LinkedList<TrieNode> list, TrieNode node) {
+    for (TrieNode n : list) {
+      if (n.getRow() == node.getRow() && n.getColumn() == node.getColumn() && n.getValue() ==
+          node.getValue()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Checks if a list contains the given string.
+   *
+   * @param list the list to be searched in
+   * @param word the word to be searched for in the list
+   * @return true if the list contains the string, false otherwise
+   */
+  private boolean containsWord (List<String> list, String word) {
+    for (String w : list) {
+      if (w.equals(word)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Checks if two matrices of trie nodes are equal.
+   *
+   * @param grid1 the first matrix to be compared
+   * @param grid2 the second matrix to be compared
+   * @return true if they contains identical elements, false otherwise
+   */
+  private boolean equalMatrices (TrieNode[][] grid1, TrieNode[][] grid2) {
```

Comment:
This could be done with Arrays.equals https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#equals(java.lang.Object[],%20java.lang.Object[]) if you added a custom equals() method to your TrieNode. This would be a very clean way of doing it.